packages/react-components/react-virtualizer/src/components/Virtualizer/useVirtualizer.ts

```diff
@@ -0,0 +1,438 @@
+    const beforeIndexValue = childProgressiveSizes.current[iBefore];
+    if (scrollPos <= afterIndexValue && scrollPos >= beforeIndexValue) {
+      /* We've found our index - if we are exactly matching before/after index that's ok,
+      better to reduce checks if it's right on the boundary. */
+      return midpoint;
+    }
+
+    if (indexValue > scrollPos) {
+      return findIndexRecursive(scrollPos, lowIndex, midpoint - 1);
+    } else {
+      return findIndexRecursive(scrollPos, midpoint + 1, highIndex);
+    }
+  };
+
+  const getIndexFromSizeArray = (scrollPos: number): number => {
+    /* Quick searches our progressive height array */
+    if (
+      scrollPos === 0 ||
+      childProgressiveSizes.current.length === 0 ||
+      scrollPos <= childProgressiveSizes.current[0]
+    ) {
+      // Check start
+      return 0;
+    }
+
+    if (scrollPos >= childProgressiveSizes.current[childProgressiveSizes.current.length - 1]) {
+      // Check end
+      return childProgressiveSizes.current.length - 1;
+    }
+
+    return findIndexRecursive(scrollPos, 0, childProgressiveSizes.current.length - 1);
+  };
+
+  const getIndexFromScrollPosition = (scrollPos: number) => {
+    if (!getItemSize) {
+      return Math.round(scrollPos / itemSize);
+    }
+
+    return getIndexFromSizeArray(scrollPos);
+  };
+
+  const calculateTotalSize = () => {
+    if (!getItemSize) {
+      return itemSize * numItems;
+    }
+
+    // Time for custom size calcs
+    return childProgressiveSizes.current[numItems - 1];
+  };
+
+  const calculateBefore = () => {
+    if (!getItemSize) {
+      // The missing items from before virtualization starts height
+      return virtualizerStartIndex * itemSize;
+    }
+
+    if (virtualizerStartIndex <= 0) {
+      return 0;
+    }
+
+    // Time for custom size calcs
+    return childProgressiveSizes.current[virtualizerStartIndex - 1];
+  };
+
+  const calculateAfter = () => {
+    if (numItems === 0) {
+      return 0;
+    }
+
+    const lastItemIndex = Math.min(virtualizerStartIndex + virtualizerLength, numItems - 1);
+    if (!getItemSize) {
+      // The missing items from after virtualization ends height
+      const remainingItems = numItems - lastItemIndex - 1;
+      return remainingItems * itemSize;
+    }
+
+    // Time for custom size calcs
+    return childProgressiveSizes.current[numItems - 1] - childProgressiveSizes.current[lastItemIndex];
+  };
+
+  const updateChildRows = (newIndex: number) => {
+    if (numItems === 0) {
+      /* Nothing to virtualize */
+
+      return [];
+    }
+
+    if (childArray.current.length !== numItems) {
+      childArray.current = new Array(virtualizerLength);
+    }
+    const actualIndex = Math.max(newIndex, 0);
+    const end = Math.min(actualIndex + virtualizerLength, numItems);
+
+    for (let i = actualIndex; i < end; i++) {
+      childArray.current[i - actualIndex] = renderChild(i);
+    }
+  };
+
+  const setBeforeRef = (element: HTMLDivElement) => {
```

Comment:
you should wrap call back refs in ` useCallback`  otherwise they will be invoked more times than you might want